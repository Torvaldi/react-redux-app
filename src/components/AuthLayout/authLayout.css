.authBlock {
    display: flex;
    flex-direction: column;
    width: 35%;
    justify-content: center;
    align-items: center;
    background-color: var(--darkBlue);
}

.authBlock_left {
    width: 60%;
}

.authLayout {
    height: 100%;
    display: flex;
}

.imageLayout {
    width: 65%;
}

/* tablet */
@media only screen and (max-width: 1024px) {
    .authLayout {
      flex-direction: column;
      justify-content: center;
    }
    .imageLayout {
        width: 100%;
        display: none;
    }
    .authBlock {
        width: 100%;
        height: 100%;
    }
}