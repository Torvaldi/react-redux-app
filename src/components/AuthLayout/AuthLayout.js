import React from 'react';

import './authLayout.css';
import BlockText from '../BlockText/BlockText';
import propTypes from 'prop-types';

const AuthLayout = (props) => {
  return (
    <section className="authLayout" >
          <article className="authBlock">
            <article class="authBlock_left">
              <BlockText text={props.text} />
              {props.left}
            </article>
          </article>
          <article className="imageLayout">
            {props.right}
          </article>
    </section>
  );
}

AuthLayout.propTypes = {
  text: propTypes.string.isRequired,
  left: propTypes.element.isRequired,
  right: propTypes.element.isRequired,
}

export default AuthLayout;
