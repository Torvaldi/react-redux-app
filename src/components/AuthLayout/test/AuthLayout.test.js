import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import AuthLayout from '../AuthLayout';

configure({ adapter: new Adapter() });

// required props
let element;
let text;
beforeEach(() => {
    element = <React.Fragment></React.Fragment>;
    text = 'hi there';
});


describe('<AuthLayout />', () => {
    // snapshot
    it('should render Auth Componenent', () => {
        const wrapper = shallow(<AuthLayout left={element} right={element} text={text} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    // props
    it('should render with props left', () => {
        const AuthLayoutComponent = mount(<AuthLayout text={text} right={element} left={element} />);

        expect(AuthLayoutComponent.prop('left')).toEqual(element);
        expect(AuthLayoutComponent.prop('right')).toEqual(element);
        expect(AuthLayoutComponent.prop('text')).toEqual(text);
    });
    
});