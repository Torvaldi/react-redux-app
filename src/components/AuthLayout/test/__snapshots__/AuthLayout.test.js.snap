// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthLayout /> should render Auth Componenent 1`] = `ShallowWrapper {}`;
